### PR TITLE
Add Click() to GuiDriver

### PR DESCRIPTION
**Pull request by simonwhitaker**
- **PR Description**

This will allow us to test mouse clicks in integration tests, like this:

```go
// given t, of type *TestDriver
t.Views().SomeView().Click(0, 0)
```

For an example of usage, see the test added in https://github.com/jesseduffield/lazygit/pull/2899

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

### CODE REVIEW

**Review by jesseduffield — COMMENTED** [review]
This is great, I appreciate the effort you've put into this. Just the one comment

File: pkg/integration/components/test_driver.go [hunk 1]
```diff
@@ -10,18 +10,20 @@ import (
 )
 
 type TestDriver struct {
-	gui          integrationTypes.GuiDriver
-	keys         config.KeybindingConfig
-	pushKeyDelay int
+	gui             integrationTypes.GuiDriver
+	keys            config.KeybindingConfig
+	pushKeyDelay    int
+	mouseClickDelay int
```
[the remark this answers is not in the document]
Comment (simonwhitaker) [reply pkg/integration/components/test_driver.go]: Done!

**Review by jesseduffield — APPROVED** [review]
LGTM. Feel free to squash your commits

File: pkg/integration/components/view_driver.go [hunk 2]
```diff
@@ -403,6 +403,14 @@ func (self *ViewDriver) PressFast(keyStr string) *ViewDriver {
 	return self
 }
 
+func (self *ViewDriver) Click(x, y int) *ViewDriver {
+	offsetX, offsetY, _, _ := self.getView().Dimensions()
+
+	self.t.click(offsetX+1+x, offsetY+1+y)
```
Comment (simonwhitaker) [inline pkg/integration/components/view_driver.go:409]: @jesseduffield I was unsure why the additional `+1`s are needed here, but they are. I expected `offsetX+x, offsetY+y` to work here. Just curious what the intuition is that I'm missing here.

Comment (jesseduffield) [reply pkg/integration/components/view_driver.go:409]: I assume that's due to the frame of the view. Although if it also applies for views without frames (e.g. the information view) then I too am confused.

**Comment by simonwhitaker** [comment]
Commits squashed. Over to you, @jesseduffield 😺

**Pull request merged**